pkg/common/common.go

```diff
@@ -62,3 +62,21 @@ func Contains(slice []string, target string) bool {
 	}
 	return false
 }
+
+// MergeMapStringString Merge desired into existing.
+func MergeMapStringString(existing *map[string]string, desired map[string]string) bool {
```

Comment:
Right, it is not thread-safe even though it is stateless.

It should not be an issue in the controllers context. One of the input params is the pointer to a reference to k8s object annotations. Two controller threads accessing the same object concurrently *should* (AFAIK) always hold two different copies of the same object. The race condition conflict on concurrent updates is handled (resolved) by the k8s API server.

I will leave a comment just in case we see race condition issues updating same memory space by two threads.